src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -193,4 +193,39 @@ object BoringUtils {
     sinks.foreach(addSink(_, genName, true, true))
     genName
   }
+
+  /** Access a source [[Data]] that may or may not be in the current module.  If
+    * this is in a child module, then create ports to allow access the
+    * requrested source.
+    */
+  def bore(source: Data): Data = {
+    val context = Builder.captureContext()
+    val thisModule = context.currentModule.get
```

Comment:
Conceptually, do you intend this to work on an instance or a definition? if I call val copy = Instance(thisModule.toDefinition) after this boring should i expect copy to have bored ports or not?

Reply:
Good question. I put a longer response to this in the main comment area as that should make discussion easier.